design_docs/mpas_xarray_class.md

```diff
@@ -0,0 +1,185 @@
+from MpasXarray.
+
+This solution may add a level of complexity, particularly for
+developing mpas_xarray and its analysis-specific derived reader.
+It also risks confusing a new developer by having seemingly redundant
+functionality in two places.  However, it seems to me to be the
+cleanest way to satisfy both requirements.
+
+
+<h1> Design and Implementation </h1>
+
+<h2> Implementation: mpas_xarray does not include MPAS-Analysis specific
+functionality <br>
+Date last modified: 2017/02/10 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+A test branch can be found here 
+[xylar/MPAS-Analysis/variable_mapping_reorg](https://github.com/xylar/MPAS-Analysis/tree/variable_mapping_reorg)
+
+I have removed `map_variable` and `rename_variables` from `mpas_xarray`.
+I also removed any mention of the variable map from the rest of `mpas_xarray`.
+
+This branch also includes several other cleanup operations that are not
+addressing any requirements.  These include:
+   - I added a new helper function, `open_multifile_dataset`, for opening an
+     xarray data set in a single, simple command without reference to the
+     preprocessor.  This function should make opening new data sets more
+     intuitive for mpas_xarray users.  Calling the 
+     `MpasXarray.open_multifile_dataset` allows a bit more flexibility at the
+     cost of being a little bit less compact.
+   - making several utility functions non-public (it is unclear to me why anyone
+     want to call these directly):
+       - `_assert_valid_datetimes`
+       - `_assert_valid_selections`
+       - `_ensure_list`
+       - `_get_datetimes`
+       - `_remove_repeated_time_index`
+   - I have removed the ability to run `mpas_xarray.py` as a script and the associated
+     tests.  This is on the premise that 1) the test were outdated and would have
+     needed to be updated to work with the current code and 2) unit testing in
+     `test/test_mpas_xarray.py` takes care of this capability in a better way.
+   - I have tried to make variable names a bit more verbose in various places
+     (e.g. dataset in place of ds).
+   - I have tried to improve the docstrings using a syntax that should be useful
+     for generating documentation later on.
+   - I have update unit testing to work with the new inerface, notably the
+
+<h2> Implementation: MPAS-Analysis specific functionality should be supported in
+xarray preprossing <br>
+Date last modified: 2017/02/10 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+In the same branch as above, I have added two classes, `MpasXarray` in the
+`mpas_xarray` module and `MpasAnalysisXarray` in the new `dataset` module.
+The file structure is as follows:
+```
+mpas_analysis/shared/
+             -  dataset/
+                     __init__.py
+                    MpasAnalysisXarray.py
+                    utility.py
+             - mpas_xarray/
+                    __init__.py
+                    mpas_xarray.py
```

Comment:
Hmm, while I'm willing to go along with it, *I* have the same questions about why there's a separate approach.  I don't see using `mpas_xarray` outside of `MPAS-Analysis` very often if at all myself.  But I'm willing to respect your wishes to keep it autonomous.

But I'm not sure the rename to `mpas_xarray_interface.py` does much to clarify that this is the simplest reader available.  I think the directory structure alone isn't going to do that unless we renamed `mpas_xarray` to something like `mpas_simple_reader` and `dataset` to `generalized_analysis_reader` or similar.  This, of course, messes up your `mpas_xarray` branding.